src/components/pipeline-warning/pipeline-warning.js

```diff
@@ -31,58 +31,62 @@ export const PipelineWarning = ({
     return () => clearTimeout(timer);
   }, []);
 
+  const WarningMessage = ({ isVisible, title, subtitle, buttons = [] }) => {
+    if (!isVisible) {
+      return null;
+    }
+    return (
+      <div
+        className={classnames('kedro', 'pipeline-warning', {
+          'pipeline-warning--sidebar-visible': sidebarVisible,
+        })}
+      >
+        <h2 className="pipeline-warning__title">{title}</h2>
+        <p className="pipeline-warning__subtitle">{subtitle}</p>
+        <div className="pipeline-warning__button-wrapper">
+          {buttons.map((buttonProps, index) => (
+            <Button key={index} {...buttonProps} />
+          ))}
+        </div>
+      </div>
+    );
+  };
```

Comment:
Really good abstraction. You can take it even further though. This whole component can now be moved outside of `<PipelineWarning />`. If it isn't moved outside, it's at risk of re-rendering unnecessarily.

You can either move it out of this file into it's own or simply put it at the top of the component. I'd do the latter.